src/operator/aggregate/average.rs

```diff
@@ -10,21 +10,23 @@ use std::{
 };
 
 /// Intermediate representation of an average as a `(sum, count)` pair.
-#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, SizeOf)]
-pub struct Avg<T> {
+#[derive(
+    Debug, Clone, Eq, PartialEq, Ord, PartialOrd, SizeOf, bincode::Decode, bincode::Encode,
+)]
+pub struct Avg<T: bincode::Decode + bincode::Encode> {
```

Comment:
Fixed.

Reply:
> You shouldn't need to add bounds to everything in this file, just in the places where `Decode`/`Encode` are needed

Turns out the nexmark queries need Encode/Decode for Avg<T> too but instead of deriving it, I just implement it now which should lessen the type constraints a bit.